packages/cfd/src/Containers/trade-modal.tsx

```diff
@@ -0,0 +1,409 @@
+            </Popover>
+        </div>
+        <Popover
+            className='cfd-trade-modal__password-popover'
+            alignment='left'
+            message={localize('Change Password')}
+            relative_render
+            zIndex={9999}
+        >
+            <Button
+                className='cfd-trade-modal__password-action'
+                transparent
+                onClick={onClick}
+                icon={
+                    <Icon
+                        icon='IcEdit'
+                        className='da-article__learn-more-icon'
+                        custom_color='var(--text-less-prominent)'
+                    />
+                }
+            />
+        </Popover>
+    </div>
+);
+
+const mobileDownloadLink = (platform: TPlatformType, type: 'ios' | 'android' | 'huawei') => {
+    switch (platform) {
+        case 'dxtrade':
+            return getPlatformDXTradeDownloadLink(type);
+        case 'ctrader':
+            return getPlatformCTraderDownloadLink(type);
+        case 'derivez':
+            return getPlatformDerivEZDownloadLink(type);
+        default:
+            return getPlatformDXTradeDownloadLink(type);
+    }
+};
+
+const getTitle = (market_type: string, is_eu_user: boolean) => {
+    if (is_eu_user) localize('MT5 CFDs');
+    return market_type;
+};
+
+const DxtradeDesktopDownload = ({ dxtrade_tokens, is_demo }: TDxtradeDesktopDownloadProps) => {
+    return (
+        <React.Fragment>
+            <a
+                className='cfd-trade-modal__dxtrade-button'
+                href={getDXTradeWebTerminalLink(
+                    is_demo ? 'demo' : 'real',
+                    dxtrade_tokens && dxtrade_tokens[is_demo ? 'demo' : 'real']
+                )}
+                target='_blank'
+                rel='noopener noreferrer'
+            >
+                <Icon className='cfd-trade-modal__dxtrade-button-icon' icon='IcBrandDxtradeWordmark' size={36} />
+                <div className='cfd-trade-modal__dxtrade-button-text'>
+                    <Text color='colored-background' size='xxs' weight='bold'>
+                        <Localize i18n_default_text='Web terminal' />
+                    </Text>
+                </div>
+            </a>
+        </React.Fragment>
+    );
+};
+
+const DerivEZDesktopDownload = () => {
+    return (
+        <React.Fragment>
+            <a className='cfd-trade-modal__dxtrade-button' href={''} target='_blank' rel='noopener noreferrer'>
+                <Icon className='cfd-trade-modal__dxtrade-button-icon' icon='IcBrandDerivEzWordmark' size={36} />
+                <div className='cfd-trade-modal__dxtrade-button-text'>
+                    <Text color='colored-background' size='xxs' weight='bold'>
+                        <Localize i18n_default_text='Web terminal' />
+                    </Text>
+                </div>
+            </a>
+        </React.Fragment>
+    );
+};
+
+const CTraderDesktopDownload = () => {
```

Comment:
Let's also reuse this for all the components instead of duplicating it multiple times